src/ServiceControl.Config/Commands/UpgradeAuditInstanceCommand.cs

```diff
@@ -0,0 +1,125 @@
+        {
+            if (LicenseChecks)
+            {
+                var licenseCheckResult = serviceControlInstaller.CheckLicenseIsValid();
+                if (!licenseCheckResult.Valid)
+                {
+                    windowManager.ShowMessage("LICENSE ERROR", $"Upgrade could not continue due to an issue with the current license. {licenseCheckResult.Message}.  Contact contact@particular.net", hideCancel: true);
+                    return;
+                }
+            }
+
+            var instance = InstanceFinder.FindInstanceByName<ServiceControlAuditInstance>(model.Name);
+            instance.Service.Refresh();
+
+            var upgradeInfo = UpgradeControl.GetUpgradeInfoForTargetVersion(serviceControlInstaller.ZipInfo.Version, instance.Version);
+            var upgradeOptions = new ServiceControlUpgradeOptions { UpgradeInfo = upgradeInfo };
+
+            if (instance.Service.Status != ServiceControllerStatus.Stopped &&
+                !windowManager.ShowYesNoDialog($"STOP INSTANCE AND UPGRADE TO {serviceControlInstaller.ZipInfo.Version}",
+                    $"{model.Name} needs to be stopped in order to upgrade to version {serviceControlInstaller.ZipInfo.Version}.",
+                    "Do you want to proceed?",
+                    "Yes I want to proceed", "No"))
+            {
+                return;
+            }
+            
+            await UpgradeAuditInstance(model, instance, upgradeOptions);
+
+            eventAggregator.PublishOnUIThread(new RefreshInstances());
+        }
+
+        async Task UpgradeAuditInstance(InstanceDetailsViewModel model, ServiceControlAuditInstance instance, ServiceControlUpgradeOptions upgradeOptions)
+        {
+            using (var progress = model.GetProgressObject($"UPGRADING {model.Name}"))
+            {
+                var reportCard = new ReportCard();
+                var restartAgain = model.IsRunning;
+
+                var stopped = await model.StopService(progress);
+
+                if (!stopped)
+                {
+                    eventAggregator.PublishOnUIThread(new RefreshInstances());
+
+                    reportCard.Errors.Add("Failed to stop the service");
+                    reportCard.SetStatus();
+                    windowManager.ShowActionReport(reportCard, "ISSUES UPGRADING INSTANCE", "Could not upgrade instance because of the following errors:");
+
+                    return;
+                }
+
+                reportCard = await Task.Run(() => serviceControlAuditInstaller.Upgrade(instance, upgradeOptions, progress));
+
+                if (reportCard.HasErrors || reportCard.HasWarnings)
+                {
+                    windowManager.ShowActionReport(reportCard, "ISSUES UPGRADING INSTANCE", "Could not upgrade instance because of the following errors:", "There were some warnings while upgrading the instance:");
+                }
+                else
```

Comment:
Instead of `else` it seems a `return` could be part of the main if scope?

Reply:
To reduce the nesting. I think that is a good suggestion. Will fix